avocado/plugins/assets.py

```diff
@@ -0,0 +1,245 @@
+            self.asgmts[self.current_klass] = {}
+            self.generic_visit(node)
+    # visit_ClassDef()
+
+    def visit_FunctionDef(self, node):  # pylint: disable=C0103
+        """
+        Visit FunctionDef on AST and save current method.
+        :param node: AST node to be evaluated
+        :type node: ast.*
+        """
+        self.current_method = node.name
+        self.asgmts[self.current_klass][self.current_method] = {}
+        self.generic_visit(node)
+    # visit_FuctionDef()
+
+    def visit_Assign(self, node):  # pylint: disable=C0103
+        """
+        Visit Assign on AST and build list of assignments that matches the
+        pattern pattern `name = string`.
+        :param node: AST node to be evaluated
+        :type node: ast.*
+        """
+        if isinstance(node.value, ast.Str):
+            # make sure we are into a method, we are not supporting attributes
+            # and module constant assignments at this time
+            if self.current_klass and self.current_method:
+                # variables just to make dictionary assignment line shorter
+                cur_klass = self.current_klass
+                cur_method = self.current_method
+                name = node.targets[0].id
+                self.asgmts[cur_klass][cur_method][name] = node.value.s
+        self.generic_visit(node)
+    # visit_Assign()
+
+    def visit_Call(self, node):  # pylint: disable=C0103
+        """
+        Visit Calls on AST and build list of calls that matches the pattern.
+        :param node: AST node to be evaluated
+        :type node: ast.*
+        """
+        if isinstance(node.func, ast.Attribute):
+            if self.pattern in node.func.attr:
+                call = self._parse_args(node)
+                if call:
+                    self.calls.append(call)
+    # visit_Call()
+
+    def fetch_assets(self):
+        """
+        Fetches the assets based on keywords listed on self.calls.
+        :returns: list of names that were successfuly fetched and list of
+        fails.
+        """
+        cache_dirs = data_dir.get_cache_dirs()
+        success = []
+        fail = []
+        for call in self.calls:
+            expire = call.pop('expire', None)
+            if expire is not None:
+                expire = data_structures.time_to_seconds(str(expire))
+            try:
+                asset.Asset(**call, cache_dirs=cache_dirs,
+                            expire=expire).fetch()
+                success.append(call['name'])
+            except EnvironmentError as failed:
+                fail.append(failed)
+        return success, fail
+    # fetch_assets()
+# FetchAssetHandler()
+
+
+class Assets(CLICmd):
+    """
+    Implements the avocado 'assets' subcommand
+    """
+    name = 'assets'
+    description = 'Manage assets'
+
+    def configure(self, parser):
+        """
+        Add the subparser for the assets action.
+
+        :param parser: Main test runner parser.
```

Comment:
I understand this docstring is in many other plugins, but it's actually a bit misleading.  The parser is the `avocado` command line application parser, which is a lot more than a "test runner", but the main entry point into most of Avocado functionality.  I'd go with:

```
:param parser: The Avocado command line application parser
:type parser: :class:`avocado.core.parser.ArgumentParser`
```

See https://github.com/avocado-framework/avocado/pull/3339 for the resolution on the other occurrences.